processor/otel/consumer_test.go

```diff
@@ -578,6 +578,74 @@ func TestArrayLabels(t *testing.T) {
 	}, tx.Labels)
 }
 
+func TestConsumeTracesExportTimestamp(t *testing.T) {
+	traces, otelSpans := newTracesSpans()
+
+	// The actual timestamps will be non-deterministic, as they are adjusted
+	// based on the server's clock.
+	//
+	// Use a large delta so that we can allow for a significant amount of
+	// delay in the test environment affecting the timestamp adjustment.
+	const timeDelta = time.Hour
+	const allowedError = 5 // seconds
```

Comment:
I've had success testing time using a helper func to wrap `time.Now()`, so that the helper func can be redefined in tests and return a specific given time, ie.
```go
timeNow = func() time.Time { return time.Now() }
```
and then over-writing it to return a specific time in the tests, if you'd rather do that than have a delta. I'm fine with either.

Reply:
I'm not super keen on patching unexported variables, as I'd prefer to test using only the exported API. In the past I've gone all in on having a `Clock` interface which can be passed in. I'd like to avoid this as it's pretty invasive.

Let's stick with the current way and we can see how it goes, happy to come back and reassess.